### PR TITLE
Mccalluc/new style analyses urls

### LINKED ISSUE #2002: Replace analysis urls with fb2 urls with appropriate facet

### PR DESCRIPTION
**Pull request by mccalluc**
Fix #2002

### CODE REVIEW

**Comment by mccalluc** [comment]
@scottx611x : This might work, but then I don't have analyses running locally. Could you first do a sanity check, and then either deploy in your environment, or help me sort out what's needed for me to run analyses in mine?

File: refinery/core/models.py [hunk 1]
```diff
@@ -698,13 +701,31 @@ def get_studies(self, version=None):
...
+
+    def _get_assays(self, version=None):
         return Assay.objects.filter(
             study=Study.objects.filter(
                 investigation=self.get_investigation()
             )
         )
 
+    def _get_assay(self):
+        assays = self.data_set._get_assays()
```
Comment (scottx611x) [inline refinery/core/models.py]: I think `_get_assay()` could be condensed to:
```
 return Assay.objects.get(study=self.get_latest_study())
```

File: refinery/core/models.py [hunk 2]
```diff
@@ -1483,6 +1504,14 @@ def get_input_file_uuid_list(self):
             input_file_uuid_list.append(cur_fs_uuid)
         return input_file_uuid_list
 
+    def data_sets_query(self):
+        analysis_facet_name = '{}_{}_{}_s'.format(
+            NodeIndex.ANALYSIS_UUID_PREFIX,
+            self.data_set._get_study().id,
```
Comment (scottx611x) [inline refinery/core/models.py]: Is `self.data_set` redundant here? Aren't we already in the DataSet class?

**Comment by mccalluc** [comment]
TODO: manual testing.
```
from factory_boy.utils import make_analyses_with_single_dataset
make_analyses_with_single_dataset(<number>, <user_instance>)
```

File: refinery/core/models.py [hunk 3]
```diff
@@ -698,13 +701,31 @@ def get_studies(self, version=None):
             investigation=self.get_investigation(version)
         )
 
-    def get_assays(self, version=None):
+    def _get_study(self):
```
Comment (mccalluc) [inline refinery/core/models.py]: yep. Thanks!

File: refinery/core/models.py (hunk 1 above)
Comment (mccalluc) [inline refinery/core/models.py]: I created a new `get_latest_assay`

File: refinery/core/models.py (hunk 2 above)
Comment (mccalluc) [inline refinery/core/models.py]: No, this is correct: We're in [`Analysis`](https://github.com/refinery-platform/refinery-platform/pull/2009/files/f710cd4a4b2a9695155940522498c730c85d34d2#diff-10b1a3c979459ad2810c191d2321daa3L1121) and need to go up to get the `DataSet`

**Comment by mccalluc** [comment]
IGV failure. Still want to do manual tests.

**Comment by mccalluc** [comment]
@scottx611x : I think this is ready to go: The new style url does come through when I hard reload. Approve and I'll merge.

screen shot 2017-08-18 at 11 56 29 am

File: refinery/core/tests.py [hunk 4]
```diff
@@ -1501,6 +1501,13 @@ def test_galaxy_tool_file_import_state_is_empty_without_import_progress(
             []
         )
 
+    def test_data_sets_query(self):
+        self.assertRegexpMatches(
+            self.analysis_with_node_analyzed_further.data_sets_query(),
+            '%7B%22REFINERY_ANALYSIS_UUID_\d+_\d+_s%22%3A%20%.+%22%7D'
+            # TODO: UUID_RE when that is merged
```
Comment (scottx611x) [inline refinery/core/tests.py]: Do this TODO

**Pull request merged**